apps/web/src/components/docs/Docs.tsx

```diff
@@ -22,29 +24,69 @@ const getMDXComponent = mdxBundler.getMDXComponent;
 
 const DOCS_WRAPPER_ELEMENT_ID = 'embedded-docs';
 
+const globalOverrides = {
+  frameworkterminal: {
+    FrameworkTerminal: () => <nv-framework-terminal></nv-framework-terminal>,
+  },
+  framework: () => null,
+  echoterminal: {
+    EchoTerminal: () => <nv-framework-terminal></nv-framework-terminal>,
+  },
+};
+
 type DocsProps = PropsWithChildren<
   DocsQueryResults & {
     isLoading?: boolean;
-    actions: ReactNode;
+    actions?: ReactNode;
+    isChildDocs?: boolean;
+    mappings: Record<string, Record<string, string> | string>;
   }
 >;
 
 /*
  * Render the mdx for our mintlify docs inside of the web.
  * Fetching the compiled mdx from another service and then try to map the markdown to react components.
  */
-export const Docs = ({ code = '', description = '', title = '', isLoading, children, actions }: DocsProps) => {
+export const Docs = ({
+  code = '',
+  description = '',
+  title = '',
+  isLoading,
+  children: docsChildren,
+  actions = null,
+  isChildDocs = false,
+  mappings = {},
+}: DocsProps) => {
   const Component = useMemo(() => {
     if (code.length === 0) {
       return null;
     }
 
+    const globals = Object.keys(mappings).reduce((prev: Record<string, Record<string, any> | any>, key) => {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** I recommend "Component" to be a separate, named component outside of the Docs scope that accepts props, and this reduce should be a separate helper function that lives outside the component and _could‏_ be tested in isolation